unblob/handlers/filesystem/ubi.py

```diff
@@ -95,14 +97,26 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
         )
 
 
+class UBIExtractor(Command):
```

Comment:
What if we don't subclass `Command`, but treat ubireader as a library and copy the minimum required code from `scripts/ubireader_extract_images` here? This would be a little bit more work, but I suspect the output would be more stable over time as unblob would control it completely. It would also result in the final structure we want instead of moving things around and deleting files.

Reply:
It's an easy fix to get closer to a PyPi release ((it's actually an internal proposal)) so that we can spend our time working on a UBI/UBIFS extractor using unblob's API and utilities later on rather than maintaining a standalone vendored script.

Reply:
So you want to completely remove `ubi-reader` as a dependency long-term?

Reply:
> So you want to completely remove `ubi-reader` as a dependency long-term?

Probably yeah.

Reply:
> I'm not suggesting vendoring, I'm suggesting having our own siomplified CLI to the ubi-reader library instead of using ubireader_extract_images at all.

I looked into it and there's a problem with this approach. If ubi-reader encounter an error and that `settings.error_action` is set to `exit` or that the error level is considered `fatal`, ubi-reader just call `sys.exit(1)` and unblob hangs there.

See https://github.com/jrspruitt/ubi_reader/blob/master/ubireader/debug.py#L36

The advantage of calling the CLI is that it can fail or return some error without affecting unblob's processing.

Reply:
Yes I seen that too, I would just put it in extractor folder and call it as a command, but if ubi-reader will be removed, doesn't really matter.